pkg/reconciler/ingress/lister.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ingress
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/sets"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"knative.dev/networking/pkg/apis/networking/v1alpha1"
+	"knative.dev/networking/pkg/status"
+
+	"knative.dev/net-gateway-api/pkg/reconciler/ingress/config"
+)
+
+const (
+	// HTTPPortExternal is the port for external availability.
+	HTTPPortExternal = "8080"
+	// HTTPPortInternal is the port for internal availability.
+	HTTPPortInternal = "8081"
+	// HTTPSPortExternal is the port for external HTTPS availability.
+	HTTPSPortExternal = "8443"
+)
+
+func NewProbeTargetLister(logger *zap.SugaredLogger, endpointsLister corev1listers.EndpointsLister) status.ProbeTargetLister {
+	return &gatewayPodTargetLister{
+		logger:          logger,
+		endpointsLister: endpointsLister,
+	}
+}
+
+type gatewayPodTargetLister struct {
+	logger          *zap.SugaredLogger
+	endpointsLister corev1listers.EndpointsLister
+}
+
+func (l *gatewayPodTargetLister) ListProbeTargets(ctx context.Context, ing *v1alpha1.Ingress) ([]status.ProbeTarget, error) {
+	gatewayConfig := config.FromContext(ctx).Gateway
+	ns, name, _ := cache.SplitMetaNamespaceKey(gatewayConfig.LookupService(v1alpha1.IngressVisibilityClusterLocal))
+
+	eps, err := l.endpointsLister.Endpoints(ns).Get(name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get internal service: %w", err)
+	}
+
+	var readyIPs []string
```

Comment:
Nit: You could build these into a `sets.String` directly and pass it that way. Seems like its only usage is going to be concerting into a set anyway.